internal/services/network/virtual_network_gateway_resource.go

```diff
@@ -657,8 +657,8 @@ func resourceVirtualNetworkGatewayCreateUpdate(d *pluginsdk.ResourceData, meta i
 
 	id := parse.NewVirtualNetworkGatewayID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
 
+	existing, err := client.Get(ctx, id.ResourceGroup, id.Name)
```

Comment:
This will not handle the value for `err` in the update case, can you take a look at implementing differently to make sure error checking is done in both cases?

Reply:
Updated